### PR TITLE
kubeadm: Don't match DNS versions to K8s versions

### LINKED ISSUE #870: dns - GetDNSVersion does not depend of the version
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

Reopening the issue from here: https://github.com/kubernetes/kubernetes/issues/64454

## Versions

**kubeadm version** (use `kubeadm version`):
 last one - master for 1.11

## What happened?
From that Review : https://github.com/kubernetes/kubernetes/pull/64258

in this piece of code : https://github.com/kubernetes/kubernetes/pull/64258/files/8d8b47596e6e71ae705ed2fd1e946e32a98a42f0#diff-6f6577e723201d3cf7ff85de45a73068

```
const (
	kubeDNSVersion = "1.14.10"
	coreDNSVersion = "1.1.3"
)

// GetDNSVersion returns the right kube-dns version for a specific k8s version
func GetDNSVersion(kubeVersion *version.Version, dns string) string {
	// v1.9.0+ uses kube-dns 1.14.10, if feature gate "CoreDNS" is disabled.
	// v1.9.0+ uses CoreDNS  1.1.3.

	// In the future when the version is bumped at HEAD; add conditional logic to return the right versions
	// Also, the version might be bumped for different k8s releases on the same branch
	switch dns {
	case kubeadmconstants.CoreDNS:
		// return the CoreDNS version
		return coreDNSVersion
	default:
		return kubeDNSVersion
	}
}
```
there is a comment:

>// In the future when the version is bumped at HEAD; add conditional logic to return the right versions
>// Also, the version might be bumped for different k8s releases on the same branch

Obviously, there is no implementation of logic to return the DNS version that as pushed for one version of Kubernetes. (neither for kube-dns, neither for CoreDNS)

for instance:
CoreDNS 1.0.0 was pushed in the release k8s v1.9.0
CoreDNS 1.0.6 was pushed in the release k8s v1.10.0
CoreDNS 1.1.3 is now pushed in the coming release k8s v1.11.0

However, the last version CoreDNS 1.1.3 is compatible with all versions of k8s (1.9 and highers).

## What you expected to happen?
A better description on what logic should be encoded in this function. Based on example of above, should that GetDNSVersion function return always 1.1.3 , or check k8s version and return 1.0.0 or 1.0.6 or 1.1.3 ?

Maybe explanation on when this function will not be called with the version of k8s it is released on.

/sig cluster-lifecycle
/cc @luxas
/cc @fturib

### PR DESCRIPTION
**Pull request by rosti**
**What this PR does / why we need it**:

Some code in kubeadm was designed with the intent, that in the future CoreDNS
and kube-dns versions will match to specific K8s versions. This code is not
functional, since it does not perform any version matching. As of this moment,
no version matching is planned and a lot of boilerplate code is left useless.
The solution is simple - remove the unneeded parts to simplify the flow.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#870

**Special notes for your reviewer**:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/assign @luxas
/assign @timothysc
/kind cleanup

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fturib** [comment]
@rajansandeep : can you review here and double check there will be no impact for CoreDNS deployment/upgrade using kubeadm ? Thanks.

File: cmd/kubeadm/app/phases/addons/dns/dns.go [hunk 1]
```diff
@@ -101,14 +96,13 @@ func kubeDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interfac
...
 
-	// Get the YAML manifest conditionally based on the k8s version
-	kubeDNSDeploymentBytes := GetKubeDNSManifest(k8sVersion)
-	dnsDeploymentBytes, err := kubeadmutil.ParseTemplate(kubeDNSDeploymentBytes,
+	// v1.8.0+ has only one known YAML manifest spec for KubeDNS
+	dnsDeploymentBytes, err := kubeadmutil.ParseTemplate(KubeDNSDeployment,
 		struct{ ImageRepository, Arch, Version, DNSBindAddr, DNSProbeAddr, DNSDomain, MasterTaintKey string }{
 			ImageRepository: cfg.ImageRepository,
 			Arch:            runtime.GOARCH,
 			// Get the kube-dns version conditionally based on the k8s version
```
Comment (chuckha) [inline cmd/kubeadm/app/phases/addons/dns/dns.go]: This comment should be removed

**Review by chuckha — COMMENTED** [review]
I'm very happy about this change, it simplifies this workflow greatly. 👏 

I want to check my understanding to make sure this is what I think it is: kubedns and coredns both do not depend on the k8s version. In this case 1.14.10 works for 1.9, 1.10 and 1.11 and when the version of coredns gets bumped to say, 1.15, it will work with all of the supported versions of k8s? Is this documented somewhere or are you in the core/kubeDNS community?

I want to double check this because we've had a lot of upgrade issues in the past cycle and want to make sure we don't run into more.

Thank you for this change!

**Comment by rosti** [comment]
This PR does not change the behavior of kubeadm in any way. It is a simple cleanup. The CoreDNS/kube-dns versions are known to work with the supported by kubeadm Kubernetes versions.
Considering this and kubernetes/kubeadm#870, in a short discussion on Slack with @luxas I decided to do this PR.
On the question of documentation, I am not sure if this behavior is documented somewhere currently. Yet, it is the current behavior and if it isn't documented, then we should do so.

**Review by chuckha — APPROVED** [review]
thanks for the update! lgtm

**Comment by rosti** [comment]
Removed the old 1.8 commend

**Comment by rosti** [comment]
Fixed a couple of other comments, that were left over.

**Comment by dixudx** [comment]
@rosti Very good cleanup. 

/lgtm

**Comment by timothysc** [comment]
@luxas what do you think here?

**Review by luxas — APPROVED** [review]
@timothysc I think this is valuable to get into v1.11.
We had this for a reason, the manifest changed frequently for different k8s versions. Now things have settled and we're using coredns as the default which my impression is kinda stable, so now we can remove the unnecessary conditionals here IMO.

/lgtm
/approve

**Pull request merged**